### PR TITLE
Allow applicationId to be specified on build

### PR DESCRIPTION
**Pull request by shockdesign**
Added the ability to define a specific applicationId (in case it's different to the package name)

<details>
  Thanks for submitting a PR! Please read these instructions carefully:

  - [x] Explain the **motivation** for making this change.
  - [ ] Provide a **test plan** demonstrating that the code is solid.
  - [x] Match the **code formatting** of the rest of the codebase.
  - [x] Target the `master` branch, NOT a "stable" branch.

  Please read the [Contribution Guidelines](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md) to learn more about contributing to React Native.
</details>

## Motivation (required)

We have a specific requirement that our package name differs from the applicationId but not in a way that allows us to add it as a suffix to our package name. I wanted to add the option to specify a specific applicationId.

## Test Plan (required)

`react-native run-android --appId <Specific Application Id>`

### CODE REVIEW

**Comment by shockdesign** [comment]
@facebook-github-bot It looks like Travis CI is failing on the iOS tests which are timing out. Note, this change has nothing to do, nor should it affect the iOS builds.

**Comment by henrikra** [comment]
So basically what you mean is that when running `react-native run-android` react native assumes that you `applicationId` inside `build.gradle` is the same as `package` in `AndroidManifest.xml` right?

So with this pr I could run `react-native run-android --appId com.company.example` as an example?

If yes this pr is great :)

One suggestion is that would it make sense that the script would read the `applicationId` straight from `build.gradle` since that is what it has to match :)

**Comment by shockdesign** [comment]
@henrikra Correct, only really useful if your applicationId is different from your package name.

I did look into that but it would require a lot more smarts especially if you're running different flavors as it would have to know which one to pick. This seemed like a quick and easy approach to solve the problem.

**Comment by henrikra** [comment]
I think this should be merged then :)

File: local-cli/runAndroid/runAndroid.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ function buildAndRun(args) {
       'utf8'
     ).match(/package="(.+?)"/)[1];
 
-  const packageNameWithSuffix = args.appIdSuffix ? packageName + '.' + args.appIdSuffix : packageName;
+  const packageNameWithSuffix = args.appId ? args.appId : args.appIdSuffix ? packageName + '.' + args.appIdSuffix : packageName;
```
Comment (shergin) [inline local-cli/runAndroid/runAndroid.js]: Could you please make it more clear?

Comment (henrikra) [inline local-cli/runAndroid/runAndroid.js]: Yeah maybe function like `getPackageNameWithSuffix({appId, appIdSuffix, packageName})`

Comment (henrikra) [inline local-cli/runAndroid/runAndroid.js]: And maybe use template literals liks `´${packageName}.${args.appIdSuffix}´`

Comment (shergin) [inline local-cli/runAndroid/runAndroid.js]: I am okay with anything which makes the code looks obvious. Even adding parentheses and indentations for original solution is enough.

**Comment by shockdesign** [comment]
@shergin Just let me know if the changes are sufficient?

**Comment by shockdesign** [comment]
@henrikra / @shergin - anything else we can do to progress this along?

**Comment by henrikra** [comment]
For me it is okay 👍

**Pull request closed without merging**